File: RZTransitions-Demo/ReadMeSnippets.m
//
//  ReadMeSnippets.m
//  RZTransitions-Demo
//
//  Created by Eric Slosser on 11/24/15.
//  Copyright © 2015 Raizlabs. All rights reserved.
//

#import "RZTransitions.h"
#import "RZSimpleCollectionViewController.h"

@interface ReadMeSnippets : UIViewController <RZTransitionInteractionControllerDelegate>

// 5
@property (nonatomic, strong) id<RZTransitionInteractionController> presentInteractionController;

@end

@implementation ReadMeSnippets

-(void)snippets
{
    // 1
    id<RZAnimationControllerProtocol> presentDismissAnimationController = [[RZZoomAlphaAnimationController alloc] init];
    id<RZAnimationControllerProtocol> pushPopAnimationController = [[RZCardSlideAnimationController alloc] init];
    [[RZTransitionsManager shared] setDefaultPresentDismissAnimationController:presentDismissAnimationController];
    [[RZTransitionsManager shared] setDefaultPushPopAnimationController:pushPopAnimationController];

    // 2
    [self setTransitioningDelegate:[RZTransitionsManager shared]];
    UIViewController *nextViewController = [[UIViewController alloc] init];
    [nextViewController setTransitioningDelegate:[RZTransitionsManager shared]];
    [self presentViewController:nextViewController animated:YES completion:nil];

    // 3
    UINavigationController *navigationController = [[UINavigationController alloc] init];
    [navigationController setDelegate:[RZTransitionsManager shared]];

    // 4
    [[RZTransitionsManager shared] setAnimationController:[[RZZoomPushAnimationController alloc] init]
                                       fromViewController:[self class]
                                         toViewController:[RZSimpleCollectionViewController class]
                                                forAction:RZTransitionAction_PushPop];

}

// 5
//@property (nonatomic, strong) id<RZTransitionInteractionController> presentInteractionController;

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Create the presentation interaction controller that allows a custom gesture
    // to control presenting a new VC via a presentViewController
    self.presentInteractionController = [[RZVerticalSwipeInteractionController alloc] init];
    [self.presentInteractionController setNextViewControllerDelegate:self];
    [self.presentInteractionController attachViewController:self withAction:RZTransitionAction_Present];
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    //  Use the present interaction controller for presenting any view controller from this view controller
    [[RZTransitionsManager shared] setInteractionController:self.presentInteractionController
                                         fromViewController:[self class]
                                           toViewController:nil
                                                  forAction:RZTransitionAction_Present];
}
// end of 5

@end

File: RZTransitions-Demo/ReadMeSnippets.swift
//
//  ReadMeSnippets.swift
//  RZTransitions-Demo
//
//  Created by Eric Slosser on 11/24/15.
//  Copyright © 2015 Raizlabs. All rights reserved.
//

import UIKit

class Foo : UIViewController, RZTransitionInteractionControllerDelegate
{
    //5
    var presentInteractionController : RZTransitionInteractionController?

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }

    func snippets()
    {
        // 1
        RZTransitionsManager.shared().defaultPresentDismissAnimationController = RZZoomAlphaAnimationController()
        RZTransitionsManager.shared().defaultPushPopAnimationController = RZCardSlideAnimationController()

        // 2
        self.transitioningDelegate = RZTransitionsManager.shared()
        let nextViewController = UIViewController()
        nextViewController.transitioningDelegate = RZTransitionsManager.shared()
        self.presentViewController(nextViewController, animated:true) {}


        // 3
        let navigationController = UINavigationController()
        navigationController.delegate = RZTransitionsManager.shared()

        // 4
        RZTransitionsManager.shared().setAnimationController( RZZoomPushAnimationController(),
            fromViewController:self.dynamicType,
            toViewController:RZSimpleCollectionViewController.self,
            forAction:.PushPop)
    }
    // 5
    override func viewDidLoad() {
        super.viewDidLoad()

        self.presentInteractionController = RZVerticalSwipeInteractionController()
        if let vc = self.presentInteractionController as? RZVerticalSwipeInteractionController {
            vc.nextViewControllerDelegate = self
            vc.attachViewController(self, withAction:.Present)
        }
    }

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        RZTransitionsManager.shared().setInteractionController( self.presentInteractionController,
            fromViewController:self.dynamicType,
            toViewController:nil,
            forAction:.Present)
    }


}
